### ISSUE TITLE
Superobing of PACE synthetic observations

### ISSUE THREAD

**Issue by guillaumevernieres** [NOAA-EMC/godas#432]
- [ ] Investigate the possibility of applying the existing superobing tools to the ioda processed PACE synthetic observations.

- [ ] If the above is not feasible, propose an alternative solution. 

-  [ ] Document the results ~with the calculation of mean and standard deviation to compare original  observations and superobs.~ Sorry, I don't understand what you wrote @mjhossen . What I meant is to look at the mean and stddev within the 1/4 deg bins that the superobing tool is using.

**Comment by mjhossen** [comment]
In this issue, I aim to create a PT10M r2d2 database for PACE obs using 10 min data after ioda-conversion. As the files are very large, I create superobing data using the existing script in the soca-science workflow. 
There are several steps for completing superobing.

Step 1: Convert PACE obs into ioda format
Step 2: Store ioda-converted data in a local r2d2 database
Step3: Finally superob the data.

I completed superobing successfully and store them in the folder: /work2/noaa/ng-godas/jhossen/pace_out/superob 

The superobs files have netcdf format issues. When I tried to read the files, they fail. I guess, they will also fail when read in the experiment

I guess this could be the issue of the machine config file.

To figure out this problem and solve it, I need more time and effort.

**Comment by guillaumevernieres** [comment]
1 - I thought the PACE observations were already in R2D2?
2 - If you were able to superob, please give us a rough idea of the execution time, the number of obs tested with, etc ...
3 - You were asked to look at the binning information, which is the output of the superobing tool that you have been using ... Mean and stddev of obs in each bins. Does that make sense?

**Comment by mjhossen** [comment]
Thanks @guillaumevernieres. I will do that.

**Comment by mjhossen** [comment]
I put the information in a doc file: https://docs.google.com/document/d/1xZKK1JrBIfbphHTKio7447PHhzYX29GFRxOuHYCloMw/edit

**Comment by guillaumevernieres** [comment]
> I put the information in a doc file: https://docs.google.com/document/d/1xZKK1JrBIfbphHTKio7447PHhzYX29GFRxOuHYCloMw/edit

Yes, it should save the stddev. It is added to the original error:
https://github.com/JCSDA-internal/soca-science/blob/2739ac29a49110fa8cdb8d1129a2a4b08e0c20b4/tools/obs_superob/superob.cc#L404

**Comment by mjhossen** [comment]
> 
1. PACE obs data has not been pushed to the r2d2 database ( I checked today).
2. The rough time is taken about 49 sec with the number of obs tested: 4377324 (number of obs after superobing : 51301). 
3. Mean of obs in each obs bin : 85.3301. 
Stddev is calculated but not saved in the superobing output. I explore the code, it seems we need to modify the code to print the Stddev.

**Comment by guillaumevernieres** [comment]
> > 
> 
> 1. PACE obs data has not been pushed to the r2d2 database ( I checked today).
> 2. The rough time is taken about 49 sec with the number of obs tested: 4377324 (number of obs after superobing : 51301).
> 3. Mean of obs in each obs bin : 85.3301.
>    Stddev is calculated but not saved in the superobing output. I explore the code, it seems we need to modify the code to print the Stddev.

What you wrote in point 3 doesn't make sense @mjhossen . The standard deviation is saved as obs error (added to the original error) and the mean of the original observations in the 1/4 deg bin becomes the obs value ... This is what that superobing/binning tool does . I'm not sure how else to explain it.

**Comment by mjhossen** [comment]
@guillaumevernieres you are right. I have rechecked. I was confused when I saw mean in the error (apologize for this confusion). 
Here is the statistics:
```
Obs_in (radiance):
mean: 14984.5
error (stddev): 124.5
Obs_out (radiance):
mean: 50.2856
error (stddev): 41.329
```
Mean is high in obs_in due to range (-3.78059e+06, 885806) 
After superobing the range is (-1528.43, 142.524)

**Comment by guillaumevernieres** [comment]
@mjhossen , re-read the issue description and my comments: We're not asking for the mean/stddev of the obs values in the files, we're asking for the mean/stddev in the 1/4 deg bins ... which is exactly the output of the superobing tool that you are using.

**Comment by mjhossen** [comment]
ObsError is constant (value is .5)

group: ObsError {
  variables:
        float radiance(nlocs, nchans) ;
                radiance:_FillValue = 999.f ;
                string radiance:units = "W m^-2 um^-1 sr^-1" ;
  data:

   radiance =
  0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,
      0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,
      0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,
      0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,
      0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,
      0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,

**Comment by guillaumevernieres** [comment]
@travissluka even wrote a README for us: https://github.com/JCSDA-internal/soca-science/blob/2739ac29a49110fa8cdb8d1129a2a4b08e0c20b4/tools/obs_superob/README.md

**Comment by mjhossen** [comment]
Thanks @guillaumevernieres. I will look at the arguments of the superobing tool and figure out how to dump the std dev in the error ...

**Comment by mjhossen** [comment]
@guillaumevernieres I think we cannot use the current superobing script for the pace obs as the script cannot handle radiance channels. I read the Readme file from the link you shared.

**Comment by mjhossen** [comment]
@guillaumevernieres We have one more option "thinning" that you do not prefer (as we discussed earlier about the thinning). If you agree, we can do ioda-conversion with thinning.

**Comment by guillaumevernieres** [comment]
Ha! Yes, I do remember that @mjhossen , I clearly did not read @travissluka 's readme, but then again, I'm not the user.
1 - What was processed then in your test?
2 - I can think of multiple ways to still be able to bin the original obs: (1) by modifying the superobing tool, or (2) by modifying what we feed it as input ...

Thinning is fine, but that does not count as added value to this project.

**Comment by mjhossen** [comment]
@guillaumevernieres 1. I think instead of considering all the channels ( total 249), it considered only one channel data and superobs them. 2. I agree with you; we can do the binning in multiple ways. I will read the code and find a feasible way do that. In the meantime, if you have some guidance, please suggest me.

**Comment by guillaumevernieres** [comment]
@mjhossen , could you push your branch with the modified superobing tool that you worked on and also provide a sample file that you processed?

**Comment by mjhossen** [comment]
@guillaumevernieres I have pushed the branch here: https://github.com/JCSDA-internal/soca-science/tree/feature/superob

**Comment by mjhossen** [comment]
@travissluka I am running the superobing in the folder: 
```/work/noaa/ng-godas/jhossen/s2s_sprint/ioda_converter/process_radiance_superobing```
Data and yaml is located in 
```/work/noaa/ng-godas/jhossen/s2s_sprint/ioda_converter/process_radiance_superobing/tmp```

**Comment by mjhossen** [comment]
@travissluka Superobing script is working for the data, like radiance that includes channels. I will do more test. Also make it work for both with and w/o Channels. 
![Screen Shot 2023-02-17 at 4 35 48 PM](https://user-images.githubusercontent.com/65732322/219803886-f9db50d6-f022-4876-9f15-3d1420c81c8a.png)
![Screen Shot 2023-02-17 at 4 35 19 PM](https://user-images.githubusercontent.com/65732322/219803888-9b324b90-4f9d-4108-b611-36512bfcefdd.png)

You can find the modified script in my branch: https://github.com/JCSDA-internal/soca-science/tree/feature/superob
